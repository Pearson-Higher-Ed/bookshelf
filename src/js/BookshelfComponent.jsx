import React from 'react';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import Bookshelf from './Bookshelf';
import msgObject from '../../translations';

export const BookshelfComponent = function ViewerComponent(paramsObj) { // eslint-disable-line import/prefer-default-export
  const intlObj = new InternationalSupport(msgObject, paramsObj.locale);

  return (<IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}>
    <Bookshelf
    	books={paramsObj.books}
      onBookClick={paramsObj.onBookClick}
      storeUPdfUrl={paramsObj.storeUPdfUrl}		
      storeBookDetails={paramsObj.storeBookDetails}
    />
  </IntlProvider>);
};
