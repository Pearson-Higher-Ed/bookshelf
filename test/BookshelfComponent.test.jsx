/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/

/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import injectTapEventPlugin from 'react-tap-event-plugin';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import msgObject from '../translations';
import BookshelfComponent from '../src/js/BookshelfComponent';
import data from '../demo/data/BookshelfData.json';

injectTapEventPlugin();
const intlObj = new InternationalSupport(msgObject, 'en');

it('renders correctly', () => {
  const tree = renderer
    .create(<MuiThemeProvider><BookshelfComponent onBookClick={function(){}} locale={intlObj.getLocale()} books={data}/></MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});




