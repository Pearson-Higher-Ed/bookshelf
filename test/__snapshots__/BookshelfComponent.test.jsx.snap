// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="bookshelf"
  role="main"
>
  <div
    className="bookshelf-body"
  >
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="Pyschology 101 cover"
          className="image"
          src="http://content.stg-openclass.com/eps/pearson-reader/api/item/f8a98cb7-ffea-4d21-8b54-827981fd679e/1/file/DonatelleMH2-080315-MJ-CM/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          Pyschology 101
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="Structural Analysis cover"
          className="image"
          src="http://content.stg-openclass.com/eps/pearson-reader/api/item/74213210-a34d-11e5-8b87-219ee60c0efb/1/file/Blake3-082715-MJ-CM-CSS/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          Structural Analysis
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="English 101 cover"
          className="image"
          src="http://content.stg-openclass.com/eps/pearson-reader/api/item/651da29d-c41d-415e-b8a4-3eafed0057db/1/file/LutgensAtm13-071415-MJ-DW/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          English 101
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="Economics cover"
          className="image"
          src="https://content.stg-openclass.com/eps/pearson-reader/api/item/8f3e17f9-bebe-463c-a55d-87b2c976c65d/1/file/WhitingLM1-062716-so/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          Economics
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="Human Anatomy cover"
          className="image"
          src="http://content.stg-openclass.com/eps/pearson-reader/api/item/9557d0fc-0c6e-47be-96a0-71b9208a70eb/1/file/Belk5wPhys-080615-MJ-CM/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          Human Anatomy
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="Nursing Conepts cover"
          className="image"
          src="http://content.stg-openclass.com/eps/pearson-reader/api/item/c6453c40-ecda-11e5-9136-03b2eeb2556b/1/file/glossary-sample_standard/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          Nursing Conepts
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book "
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <img
          alt="Music Theory cover"
          className="image"
          src="http://content.stg-openclass.com/eps/pearson-reader/api/item/32e75b80-bf9a-11e5-929d-db71ada04282/1/file/9780134063737_et2_dcusb_l1/OPS/images/cover.jpg"
        />
        <p
          className="title"
        >
          Music Theory
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
    <div
      className="book no-book-cover"
    >
      <a
        className="bookContainer"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        <span
          className="image"
        />
        <p
          className="title"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris et ultrices augue, a interdum diam. Sed dignissim augue sit amet tempor tincidunt.
        </p>
      </a>
      <a
        aria-label="More Info"
        className="info"
        onClick={[Function]}
        role="link"
        tabIndex="0"
      >
        
      </a>
    </div>
  </div>
  <div
    aria-live="assertive"
    className="reader-only"
    id="books-assert-container"
    role="alert"
  />
</div>
`;
